crowdfunding/projects/serializers.py

```diff
@@ -0,0 +1,44 @@
+from rest_framework import serializers
+
+from .models import Project, Pledge
+
+class ProjectSerializer(serializers.Serializer):
+    id = serializers.ReadOnlyField()
+    title = serializers.CharField(max_length=200) 
+    description = serializers.CharField(max_length=None)
+    goal = serializers.IntegerField()
+    image = serializers.URLField()
+    is_open = serializers.BooleanField()
+    date_created = serializers.DateTimeField()
+    owner = serializers.ReadOnlyField(source='owner_id')
+
+    def create(self,validated_data):
+        return Project.objects.create(**validated_data)
+
+    def update(self, instance, validated_data):
+        instance.title = validated_data.get('title',instance.title)
+        instance.description = validated_data.get
+        ('description', instance.description)
+        instance.goal = validated_data.get ('goal',instance.goal)
+        instance.image=validated_data.get ('image', instance.image)
+        instance.is_open = validated_data.get('is_open', instance.is_open)
+        instance.date_created = validated_data.get('date_created', instance.date_created)
+        instance.owner = validated_data.get('owner', instance.owner)
+        instance.save()
+        return instance
+'''Pledge Serializer / Form'''
+class PledgeSerializer(serializers.ModelSerializer):
+    supporter = serializers.ReadOnlyField(source='supporter.username')
+    class Meta:
+        model = Pledge
+        # fields = ['id', 'amount', 'comment','anonymous', 'project', 'supporter']
+        # read_only_fields = ['id', 'supporter']  '''THIS LINE __all__ replaces the needs in the model.serializer to  the fields '''
+        fields = '__all__'
+    def create(self, validated_data):
+        return Pledge.objects.create(**validated_data)
+    def update(self, instance, validated_data):
+        instance.save()
+        return instance
```

Comment:
I'm not sure create and update are needed here, because this should be taken care of by the serializer